x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/SqlDataTypes.java

```diff
@@ -153,23 +185,46 @@
             INTERVAL_MINUTE_TO_SECOND,
             GEO_SHAPE,
             GEO_POINT,
-            SHAPE))
+            SHAPE,
+            // arrays
+            BOOLEAN_ARRAY,
+            BYTE_ARRAY,
+            SHORT_ARRAY,
+            INTEGER_ARRAY,
+            LONG_ARRAY,
+            DOUBLE_ARRAY,
+            FLOAT_ARRAY,
+            HALF_FLOAT_ARRAY,
+            SCALED_FLOAT_ARRAY,
+            KEYWORD_ARRAY,
+            TEXT_ARRAY,
+            DATETIME_ARRAY,
+            IP_ARRAY,
+            BINARY_ARRAY,
+            GEO_SHAPE_ARRAY,
+            GEO_POINT_ARRAY,
+            SHAPE_ARRAY))
             .sorted(Comparator.comparing(DataType::typeName))
             .collect(toUnmodifiableList());
 
     private static final Map<String, DataType> NAME_TO_TYPE = TYPES.stream()
             .collect(toUnmodifiableMap(DataType::typeName, t -> t));
 
     private static final Map<String, DataType> ES_TO_TYPE = TYPES.stream()
-            .filter(e -> e.esType() != null)
-            .collect(toUnmodifiableMap(DataType::esType, t -> t));
+        .filter(e -> e.esType() != null)
+        .filter(e -> e instanceof ArrayDataType == false)
```

Comment:
those instance of can also be avoided if there is a method isArray() on the datatype (not static)